### PR TITLE
Update GitHub actions

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes https://github.com/sourcegraph/cody-issues/issues/429.

## Test plan
1. CI

Expected: 
Post Java Setup step takes shorter than 4 min 😅 
it takes below 2 min nows

### CODE REVIEW

File: src/test/kotlin/com/sourcegraph/cody/agent/protocol/PositionTest.kt [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.sourcegraph.cody.agent.protocol
 
 import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import org.junit.Ignore
 
+@Ignore
 class PositionTest : BasePlatformTestCase() {
```
Comment (mkondratek) [inline src/test/kotlin/com/sourcegraph/cody/agent/protocol/PositionTest.kt]: This is super annoying. We cannot have test like this.

Comment (odisseus) [reply src/test/kotlin/com/sourcegraph/cody/agent/protocol/PositionTest.kt]: Why? What is the problem with this test?

Comment (mkondratek) [reply src/test/kotlin/com/sourcegraph/cody/agent/protocol/PositionTest.kt]: it is falky

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**